extensions/tasks-parent/tasks-client/src/main/java/io/resys/thena/tasks/client/spi/visitors/TaskCommandVisitor.java

```diff
@@ -111,28 +144,161 @@ private Task visitCreateTask(CreateTask command) {
+        .withArchived(targetDate)
+        .withUpdated(targetDate);
+    return this.current;
+  }
+
+  private Task visitCommentOnTask(CommentOnTask command) {
+    final var comments = new ArrayList<>(current.getComments());
+    final var id = ctx.getGid().getNextId(DocumentType.TASK);
+    comments.add(ImmutableTaskComment.builder()
+        .id(id)
+        .commentText(command.getCommentText())
+        .replyToId(command.getReplyToCommentId())
+        .username(command.getUserId())
+        .created(handleTargetDate(command.getTargetDate()))
+        .build());
+    this.current = this.current
+        .withComments(comments.stream().toList());
+    return this.current;
+  }
+
+  private Task visitChangeTaskComment(ChangeTaskComment command) {
+    final var id = command.getCommentId();
+
+    final var newComment = ImmutableTaskComment.builder()
+        .id(id)
+        .commentText(command.getCommentText())
+        .replyToId(command.getReplyToCommentId())
+        .username(command.getUserId())
+        .created(handleTargetDate(command.getTargetDate()))
+        .build();
+
+    final var newCommentList = visitList(current.getComments(), newComment, id, Task.TaskComment::getId, "Comment");
+
+    this.current = this.current
+        .withComments(newCommentList.stream().sorted(Comparator.comparing(Task.TaskComment::getId)).toList());
+    return this.current;
+  }
+
+  private Task visitAssignTaskRoles(AssignTaskRoles command) {
+    this.current = this.current
+        .withRoles(command.getRoles().stream().distinct().sorted().toList())
+        .withUpdated(handleTargetDate(command.getTargetDate()));
+    return this.current;
+  }
+
+  private Task visitAssignTask(AssignTask command) {
+    this.current = this.current
+        .withAssigneeIds(command.getAssigneeIds().stream().distinct().sorted().toList())
+        .withUpdated(handleTargetDate(command.getTargetDate()));
+    return this.current;
+  }
+
+  private Task visitChangeTaskDueDate(ChangeTaskDueDate command) {
+    this.current = this.current
+        .withDueDate(command.getDueDate().orElse(null))
+        .withUpdated(handleTargetDate(command.getTargetDate()));
+    return this.current;
+  }
+
+  private Task visitChangeTaskInfo(ChangeTaskInfo command) {
+    this.current = this.current
+        .withTitle(command.getTitle())
+        .withDescription(command.getDescription())
+        .withUpdated(handleTargetDate(command.getTargetDate()));
+    return this.current;
+  }
+
+  private Task visitCreateTaskExtension(CreateTaskExtension command) {
+    final var extensions = new ArrayList<>(current.getExtensions());
+    final var id = ctx.getGid().getNextId(DocumentType.TASK);
+    extensions.add(ImmutableTaskExtension.builder()
+        .id(id)
+        .name(command.getName())
+        .type(command.getType())
+        .body(command.getBody())
+        .build());
+    this.current = this.current
+        .withExtensions(extensions.stream().toList())
```

Comment:
.stream().toList() is not useful here.  Remove this.  

extension is already a list.